src/main/java/bot/Bot.java

```diff
@@ -0,0 +1,169 @@
+            case "list":
+                return this.getTaskList();
+            case "mark":
+                return this.markTask(Integer.parseInt(tokens[1]) - 1);
+            case "unmark":
+                return this.unmarkTask(Integer.parseInt(tokens[1]) - 1);
+            case "todo":
+                return this.addTask(new Todo(tokens[1].trim()));
+            case "deadline":
+                final String[] deadlineArgs = tokens[1].split(" /by ");
+                return this.addTask(new Deadline(
+                        deadlineArgs[0].trim(),
+                        LocalDate.parse(deadlineArgs[1].trim(), Deadline.DATE_INPUT_FORMAT)));
+            case "event":
+                final String[] eventArgs = tokens[1].split(" /at ");
+                return this.addTask(new Event(eventArgs[0].trim(), eventArgs[1].trim()));
+            case "delete":
+                return this.deleteTask(Integer.parseInt(tokens[1]) - 1);
+            default:
+                throw new BotException("I'm sorry, but I don't know what that means :-(");
+            }
+        } catch (TaskServiceException ex) {
+            throw new BotException(ex.getMessage());
+        }
+    }
+
+    private boolean detectTerminationRequest(String query) {
+        return query.equals("bye");
+    }
+
+    private String greet() {
+        return this.constructResponse(Bot.GREETING);
+    }
+
+    private String bidFarewell() {
+        return this.constructResponse(Bot.FAREWELL);
+    }
+
+    private String getTaskList() throws TaskServiceException {
+        final String taskList = "Here are the tasks in your list:\n"
+                + this.constructTaskList(this.taskService.get());
+        return this.constructResponse(taskList);
+    }
+
+    private String addTask(Task newTask) throws BotException, TaskServiceException {
+        if (newTask.getDescription().equals("")) {
+            throw new BotException("The description of a task cannot be empty.");
+        }
+        this.taskService.create(newTask);
+        final String response =
+                "Got it. I've added this task:\n  " + newTask
+                        + "\nNow you have " + this.taskService.getNumberOfTasks() + " tasks in the list.";
+        return this.constructResponse(response);
+    }
+
+    private String markTask(int taskId) throws BotException, TaskServiceException {
+        final Optional<Task> result = this.taskService.getById(taskId);
+        if (result.isEmpty()) {
+            throw new BotException("Task to mark doesn't exist");
+        }
+
+        result.get().markAsDone();
+        this.taskService.update(taskId, result.get());
+
+        final String response = "Nice! I've marked this task as done:\n  " + result.get();
+        return this.constructResponse(response);
+    }
+
+    private String unmarkTask(int taskId) throws BotException, TaskServiceException {
+        final Optional<Task> result = this.taskService.getById(taskId);
```

Comment:
Maybe it would be better if the name "result" was changed to something more explicit